### PR TITLE
Add todo list to demo project frontend

### LINKED ISSUE #8: Create components for todo list ( todo-item, list-item)

### PR DESCRIPTION
**Pull request by beprajwal**
# What problem does this change will solve?
- [x] Add todo list component with deletion and toggle functionality
- [x] Setup redux with ready thunk functions for todo actions

# Link to Issue Tracker (Github Issue)
Closes #8 

# Additional Details (screenshots, code snippets etc)
image

### CODE REVIEW

File: client/src/actions/todos.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+export const FETCH_TODOS_BEGIN = 'FETCH_TASKS_BEGIN';
+export const FETCH_TODOS_SUCCESS = 'FETCH_TODOS_SUCCESS';
```
Comment (Prabeshpd) [inline client/src/actions/todos.js:4]: define this in actions in similar auth structure

File: client/src/actions/todos.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      const updatedTodos = todos.map((todo) => {
+        if (todo.id === id) {
+          return {
+            ...todo,
+            completed: !todo.completed,
+          };
+        }
+        return todo;
+      });
+      dispatch(fetchTodosSuccess(updatedTodos));
```
Comment (Prabeshpd) [inline client/src/actions/todos.js]: define this functions on reducer maintain a single state

Comment (beprajwal) [reply client/src/actions/todos.js]: This is redux thunk function it isn't necessary now but when dealing with asynchronous API it is very important so I used it for now too. And the single state is being maintained.

File: client/src/actions/todos.js (hunk 1 above)
Comment (beprajwal) [reply client/src/actions/todos.js:4]: These variables are inside actions. The only difference is I haven't used `createAction()`

File: client/src/actions/todos.js (hunk 2 above)
Comment (Prabeshpd) [reply client/src/actions/todos.js]: Redux thunk has been defined on initializing the stores and redux actions so on single components we can take over from redux state

**Review by Prabeshpd — APPROVED** [review]

**Pull request merged**